presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcessFactory.java

```diff
@@ -36,68 +41,95 @@
 
 import static java.util.Objects.requireNonNull;
 
-public class NativeExecutionTaskFactory
+public class NativeExecutionProcessFactory
```

Comment:
Why do we want to combine the factories? They seem logically separated. NativeExecutionProcess is responsible for launching the process and maintain health, more like a worker role. NativeExecutionTask is responsible for scheduling task and retrieving results from external process, more like a coordinator role.

Reply:
Initially we created this factory as a workaround to take in injected objects from GUICE framework where we did't actually need a factory for factory purpose for NativeExecutionTask. But since we used the factory pattern it will be better to follow the pattern convention: XXXFactory shall create XXX. Here AAAFactory 1) creates 2 different types AAA and BBB, 2) is stateful and has control plan logic on created instances (stop() etc). It looks more like some control to me.

Reply:
We can have a separate NativeExecutionProcessFactory (it's okay to have 2 factories) that only creates a NativeExecutionProcess (for the purpose of accepting GUICE injections) and let operator do the rest (lifecycle management etc).

Reply:
One more question: Do we need to take care of the shutdown of injected resources? Shouldn't they be managed by the framework?

Reply:
Updated to bring back the `NativeExecutionTaskFactory`